edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/ServiceIdentityTree.cs

```diff
@@ -91,33 +91,80 @@ public async Task<IList<string>> GetAllIds()
             }
         }
 
-        public async Task<Option<string>> GetAuthChain(string id)
+        public async Task<Option<string>> GetAuthChain(string targetId)
         {
-            Preconditions.CheckNonWhiteSpace(id, nameof(id));
+            Preconditions.CheckNonWhiteSpace(targetId, nameof(targetId));
 
             // Auth-chain for the acting device (when there aren't any nodes in the tree yet)
-            if (id == this.actorDeviceId)
+            if (targetId == this.actorDeviceId)
             {
                 return Option.Some(this.actorDeviceId);
             }
-            else if (id == this.actorDeviceId + "/$edgeHub")
+            else if (targetId == this.actorDeviceId + "/" + Constants.EdgeHubModuleId)
             {
-                // TODO: richma - make this better
-                return Option.Some(this.actorDeviceId + "/$edgeHub" + ";" + this.actorDeviceId);
+                return Option.Some(this.actorDeviceId + "/" + Constants.EdgeHubModuleId + ";" + this.actorDeviceId);
             }
 
             using (await this.nodesLock.LockAsync())
```

Comment:
it is able to avoid running these logic for non nested-edge scenario.

Reply:
there won't be a config differentiation between nested and non-nested when we GA.  Legacy non-nested configuration is just nested with 1 level.  So there isn't really a way to diverge the behavior at runtime.  Doing that would add more code complexity anyways, it's arguably not worthwhile from a maintanence PoV.

---
In reply to: [458495127](https://github.com/Azure/iotedge/pull/3248#discussion_r458495127) [](ancestors = 458495127)